pytorch_lightning/callbacks/pruning.py

```diff
@@ -0,0 +1,384 @@
+                raise MisconfigurationException(
+                    f"The provided parameter_names {param_name} isn't in {self.PARAMETER_NAMES} "
+                )
+
+        if isinstance(pruning_fn, str):
+            pruning_fn = pruning_fn.lower()
+            if pruning_fn not in _PYTORCH_PRUNING_FUNCTIONS:
+                raise MisconfigurationException(
+                    f"The provided pruning_fn {pruning_fn} isn't available with "
+                    f"PyTorch build-in {_PYTORCH_PRUNING_FUNCTIONS.keys()} "
+                )
+            if "unstructured" not in pruning_fn:
+                if pruning_dim is None:
+                    raise MisconfigurationException(
+                        "When requesting `structured` pruning, the `pruning_dim` should be provided."
+                    )
+                if pruning_fn == "ln_structured":
+                    if pruning_norm is None:
+                        raise MisconfigurationException(
+                            "When requesting `ln_structured` pruning, the `pruning_norm` should be provided."
+                        )
+
+                    pruning_fn = self.create_pruning_fn(pruning_fn, dim=pruning_dim, n=pruning_norm)
+                else:
+                    pruning_fn = self.create_pruning_fn(pruning_fn, dim=pruning_dim)
+            else:
+                pruning_fn = self.create_pruning_fn(pruning_fn)
+
+        else:
+            bases = getattr(pruning_fn, "__bases__", None)
+            if bases is None or bases[0] != pytorch_prune.BasePruningMethod:
+                raise MisconfigurationException(
+                    f'pruning_fn is expected to be the str in {_PYTORCH_PRUNING_FUNCTIONS.keys()} '
+                    f'or a `PyTorch BasePruningMethod`. Found: {pruning_fn}')
+
+            if not use_global_unstructured:
+                raise MisconfigurationException(
+                    '`PyTorch BasePruningMethod` is currently support only for `use_global_unstructured=True`. ')
+
+        if use_global_unstructured and pruning_fn.PRUNING_TYPE != "unstructured":
+            raise MisconfigurationException(
+                'Only "unstructured" PRUNING_TYPE supported for '
+                f"the `pruning_method`. Found method {pruning_fn} of type {pruning_fn.PRUNING_TYPE}. ")
+
+        self.pruning_fn = pruning_fn
+
+        if not (prune_on_epoch_end or prune_on_fit_end):
+            raise MisconfigurationException(
+                "The `ModelPruning` won't be triggered as currently not activated on either `epoch_end` or `fit_end`.")
+
+        self.make_pruning_permanent = make_pruning_permanent
+
+        if not isinstance(amount, (int, float, Callable)):
+            raise MisconfigurationException(
+                "amount should be provided and be either an int, a float or Callable function."
+            )
+
+        self.amount = amount
+
+    def filter_parameters_to_prune(self, parameters_to_prune: Optional[List[Tuple[nn.Module, str]]]):
```

Comment:
```suggestion
    @staticmethod
    def filter_parameters_to_prune(self, parameters_to_prune: Optional[List[Tuple[nn.Module, str]]]):
```
static?

Reply:
It is a hook. I think the user might want to make some decision based on the provided input.